components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java

```diff
@@ -167,6 +179,91 @@ Get ancestor organization ids (including itself) of a given organization. The li
+            }
+        }
+
+        if (organizationId != null) {
+            String resourceName = getResourceNameForCustomText(screen, locale);
+            TextCustomizedOrgCacheEntry valueFromCache = textCustomizedOrgCache.getValueFromCache
+                    (new TextCustomizedOrgCacheKey(organizationId, resourceName), currentTenantDomain);
+            if (valueFromCache != null) {
+                Optional<CustomText> customText =
+                        getCustomText(type, name, screen, locale, valueFromCache.getCustomTextResolvedTenant());
+                return customText.orElseThrow(
+                        () -> handleClientException(ERROR_CODE_CUSTOM_TEXT_PREFERENCE_NOT_EXISTS, getTenantDomain()));
+            }
+
+            // No cache found. Start with current organization.
+            Optional<CustomText> customText = getCustomText(type, name, screen, locale, currentTenantDomain);
+            if (customText.isPresent()) {
+                return customText.get();
+            }
+
+            try {
+                // Get the details of the parent organization and resolve the custom text preferences.
+                Organization organization = organizationManager.getOrganization(organizationId, false, false);
+                String parentId = organization.getParent().getId();
+                String parentTenantDomain = organizationManager.resolveTenantDomain(parentId);
+                int parentDepthInHierarchy = organizationManager.getOrganizationDepthInHierarchy(parentId);
+
+                while (parentDepthInHierarchy > 0) {
+                    customText =
+                            getCustomText(type, name, screen, locale, parentTenantDomain);
+                    if (customText.isPresent()) {
+                        addCustomTextResolvedOrgToCache
+                                (organizationId, resourceName, currentTenantDomain, parentTenantDomain);
+                        return customText.get();
+                    }
+
+                    /*
+                        Get ancestor organization ids (including itself) of a given organization. The list is sorted
+                        from given organization id to the root organization id.
+                     */
+                    List<String> ancestorOrganizationIds = organizationManager.getAncestorOrganizationIds(parentId);
+                    if (!ancestorOrganizationIds.isEmpty() && ancestorOrganizationIds.size() > 1) {
+                        // Go to the parent organization again.
+                        parentId = ancestorOrganizationIds.get(1);
+                        parentTenantDomain = organizationManager.resolveTenantDomain(parentId);
+                        parentDepthInHierarchy = organizationManager.getOrganizationDepthInHierarchy(parentId);
+                    } else {
+                        // Reached to the root of the organization tree.
+                        parentDepthInHierarchy = 0;
+                    }
```

Comment:
refere to the parent traversal termination in the resolveBranding method.
While operation termination condition needs to be controlled by the primary org level defined in the deployment

Reply:
Fixed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/16/commits/46898cfa3908515c0c6e320c980d4d7611c7152a